plugins/module_utils/addon_utils.py

```diff
@@ -54,9 +50,11 @@ def get_managed_cluster_addon(hub_client, addon_name: str, cluster_name: str):
 def check_managed_cluster_addon_available(managed_cluster_addon) -> bool:
     if managed_cluster_addon is None:
         return False
-    for condition in managed_cluster_addon.status.conditions:
-        if condition.type == 'Available':
-            return condition.status == 'True'
+    if "status" in managed_cluster_addon.keys():
```

Comment:
this functions is not used? I guess we can remove it

Reply:
nvm, it's used in the module to double check addon availability (e.g. if user not enabling wait)